test/controllers/api/v2/locations_controller_test.rb

```diff
@@ -335,4 +336,68 @@ def add_locations
     assert_response :success
     assert_equal 1, location.reload.location_parameters.count
   end
+
+  test "should create with comma separated name" do
+    name = "%s, %s" % [RFauxFactory.gen_alpha, RFauxFactory.gen_alpha]
+    post :create, params: { :location => { :name => name } }
+    assert_response :success
+    show_response = JSON.parse(@response.body)
+    assert !show_response.empty?
+  end
+
+  test "should create with description" do
+    post :create, params: { :location => { :name => RFauxFactory.gen_alpha,  :description => RFauxFactory.gen_alpha } }
+    assert_response :success
+    show_response = JSON.parse(@response.body)
+    assert !show_response.empty?
+  end
+
+  test "should delete location" do
+    location = FactoryBot.create(:location)
+    delete :destroy, params: { :id => location.id }
+    assert_response :success
+    assert_nil Location.find_by_id(location.id)
+  end
+
+  test "should not create with same name" do
+    # test-id: bc09acb3-9ecf-4d23-b3ef-94f24e16e6db
+    name = RFauxFactory.gen_alphanumeric
+    post :create, params: { :location => { :name => name } }
+    assert_response :success
+    post :create, params: { :location => { :name => name } }
+    assert_response :unprocessable_entity
+    assert_includes(JSON.parse(response.body)["error"]["full_messages"], "Name has already been taken")
+  end
+
+  test "should not create with non existing domain" do
+    # test-id: 5449532d-7959-4547-ba05-9e194eea495d
+    domain = rand(10000..99999)
```

Comment:
this test could fail if there was a domain with a given id, I think it would be safer if you create a domain with FactoryBot, save its id and then destroy it, that actually better simulates the race condition of two users that could cause this